conda_libmamba_solver/shards.py

```diff
@@ -469,7 +469,33 @@ def repodata_shards(url, cache: RepodataCache) -> bytes:
     Update cache state.
 
     Return shards data, either newly fetched or from cache.
+
+    In offline mode, returns cached data even if expired. If no cache exists
+    in offline mode, raises RepodataIsEmpty to signal unavailability.
     """
+    # In offline mode, return cached data if available, even if expired
+    if context.offline:
+        if cache.cache_path_shards.exists():
+            # Check if cache is stale to provide appropriate warning message
+            is_stale = cache.stale()
+            if is_stale:
+                log.warning(
```

Comment:
Not sure if logs are helpful.  A user may not care if the cache is stale or not when explicitly asking for offline mode.